src/concrete/ml/torch/hybrid_model.py

```diff
@@ -53,35 +81,107 @@ def convert_conv1d_to_linear(layer_or_module):
     return layer_or_module
 
 
+# pylint: disable-next=too-many-instance-attributes
 class RemoteModule(nn.Module):
     """A wrapper class for the modules to be done remotely with FHE."""
 
     def __init__(
         self,
-        module=None,
-        server_remote_address=None,
+        module: Optional[nn.Module] = None,
+        server_remote_address: Optional[str] = None,
+        module_name: Optional[str] = None,
+        model_name: Optional[str] = None,
+        verbose: int = 0,
     ):
         super().__init__()
-        self.private_module = module
-        self.server_remote_address = server_remote_address
-        self.calibration_data = []
+        self.private_module: Optional[nn.Module] = module
+        self.server_remote_address: Optional[str] = server_remote_address
+        self.calibration_data: List = []
         self.uid = str(uuid.uuid4())
-        self.private_q_module = None
-        self.fhe_local_mode = "disable"
-        self.client: Optional[FHEModelClient] = None
-        self.path_to_keys = None
-        self.path_to_client = None
-
-    def init_fhe_client(self, path_to_client: str, path_to_keys: str):
+        self.private_q_module: Optional[QuantizedModule] = None
+        # TODO: figure out if this is good
+        self.fhe_local_mode: FHEMode = FHEMode.CALIBRATE
+        self.clients: Dict[str, Tuple[str, FHEModelClient]] = {}
+        self.path_to_keys: Optional[Path] = None
+        self.path_to_clients: Optional[Path] = None
+        self.module_name: Optional[str] = module_name
+        self.model_name: Optional[str] = model_name
+        self.verbose = verbose
+
+    def init_fhe_client(
+        self, path_to_client: Optional[Path] = None, path_to_keys: Optional[Path] = None
+    ):  # pragma:no cover
         """Set the clients keys.
 
         Args:
             path_to_client (str): Path where the client.zip is located.
             path_to_keys (str): Path where keys are located.
+
+        Raises:
+            ValueError: if anything goes wrong with the server.
         """
-        # TODO: here we need to load fhe client.zip with FHEModelClient.
-        # Either by getting it from the server with the self.uid or
-        # directly getting it when downloading the model from HF.
+        self.path_to_clients = path_to_client
+        if self.path_to_clients is None:
+            self.path_to_clients = Path() / "clients"
+        self.path_to_clients.mkdir(exist_ok=True)
+        self.path_to_keys = path_to_keys
+        if self.path_to_keys is None:
+            self.path_to_keys = Path() / "keys"
+        self.path_to_keys.mkdir(exist_ok=True)
+
+        assert self.module_name is not None
```

Comment:
could you maybe document a bit more this part ? I'm not sure what is this `shapes_response` + what this whole part does overall

Reply:
I'll add more comments along the way, thanks for the feedback on this!

Reply:
thanks, looks better now !